tests/aws/services/sns/test_sns.py

```diff
@@ -618,6 +618,41 @@ def test_remove_permission_errors(self, snapshot, aws_client, account_id):
 
         snapshot.match("topic-not-found", e.value.response)
 
+    @markers.aws.validated
+    @pytest.mark.skipif(condition=is_sns_v1_provider(), reason="Not implemented in moto")
+    def test_data_protection_policy_crud(self, snapshot, aws_client):
+        topic_name = f"topic-{short_uid()}"
+        topic_arn = aws_client.sns.create_topic(Name=topic_name)["TopicArn"]
+
+        policy_doc = {
+            "Name": "data_protection_policy",
+            "Description": "Test Policy",
+            "Version": "2021-06-01",
+            "Statement": [
+                {
+                    "Sid": "test-statement",
+                    "DataDirection": "Inbound",
+                    "Principal": ["*"],
+                    "DataIdentifier": [
+                        "arn:aws:dataprotection:us-east-1::data-identifier/EmailAddress"
+                    ],
+                    "Operation": {"Deny": {}},
+                }
+            ],
+        }
+
+        response = aws_client.sns.put_data_protection_policy(
+            ResourceArn=topic_arn, DataProtectionPolicy=json.dumps(policy_doc)
+        )
+        snapshot.match("put-data-protection-policy", response)
+
+        response = aws_client.sns.get_data_protection_policy(ResourceArn=topic_arn)
+        snapshot.match("get-data-protection-policy", response)
+
+        # check if policy shows up in the attributes
+        response = aws_client.sns.get_topic_attributes(TopicArn=topic_arn)
+        snapshot.match("get-topic-attributes", response)
```

Comment:
nit: could we maybe call `get_topic_attributes` before calling `put_data_protection_policy` to see the before/after? I'm curious about the `EffectiveDeliveryPolicy` being created 😄 

I've given it a try locally and I can confirm that the `EffectiveDeliveryPolicy` is already present before the call, so I think the logic needs to be updated. This can be done either in this PR (add the policy at topic creation) or in another PR (add a snapshot skip for the delivery policy for now in this PR and add the policy in a follow-up PR)